protocols/imap/src/main/java/org/apache/james/imap/decode/parser/FetchCommandParser.java

```diff
@@ -81,39 +81,40 @@ private FetchData fetchRequest(ImapRequestLineReader request, boolean useUid) th
             }
             request.consumeChar(')');
             
-            
             next = nextNonSpaceChar(request);
             if (next == '(') {
                 request.consumeChar('(');
-
-                next = request.nextChar();
-                switch (next) {
-                case 'C':
-                    // Now check for the CHANGEDSINCE option which is part of CONDSTORE
-                    request.consumeWord(StringMatcherCharacterValidator.ignoreCase(CHANGEDSINCE));
-                    fetch.changedSince(request.number(true));
-                    break;
-                case 'V':
-                    // Check for the VANISHED option which is part of QRESYNC
-                    request.consumeWord(StringMatcherCharacterValidator.ignoreCase(VANISHED));
-                    fetch.vanished(true);
-                    break;
-                default:
-                    break;
-                }
-               
-                
+                parseModifier(request, fetch);
+                nextNonSpaceChar(request);
+                parseModifier(request, fetch);
+                nextNonSpaceChar(request);
```

Comment:
I get there is only maybe 2 options but that seems very arbitrary code to me.

How about sth like:
```
while (parseModifier(request, fetch)) {
  nextNonSpaceChar(request);
}
``` 
with `parseModifier` return a boolean, true as long as it finds an option and false if nothing? If no options, you would not parse twice either?

I'm not sure though, let me know what you think

Reply:
Done in https://github.com/apache/james-project/pull/909/commits/b2a8404c605192adb8b788d3f6941b68c949bab9